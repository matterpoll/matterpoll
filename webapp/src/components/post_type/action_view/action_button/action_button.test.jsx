import React from 'react';
import {shallow} from 'enzyme';

import ActionButton from 'components/post_type/action_view/action_button/action_button';

describe('components/action_button/ActionButton', () => {
    const baseProps = {
        postId: 'post_id1',
        action: {
            id: 'action_id1',
            name: 'action_name',
        },
        hasVoted: false,
        actions: {
            voteAnswer: jest.fn(),
        },
    };
    test('should match snapshot', () => {
        const wrapper = shallow(<ActionButton {...baseProps}/>);
        expect(wrapper).toMatchSnapshot();
    });
    test('should match snapshot with hasVoted', () => {
        const newProps = {
            ...baseProps,
            hasVoted: true,
        };
        const wrapper = shallow(<ActionButton {...newProps}/>);
        expect(wrapper).toMatchSnapshot();
    });
});
