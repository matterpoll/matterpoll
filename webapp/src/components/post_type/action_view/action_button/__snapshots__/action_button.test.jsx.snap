// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/action_button/ActionButton should match snapshot 1`] = `
<ButtonToolbar
  bsClass="btn-toolbar"
  style={
    Object {
      "marginLeft": 0,
    }
  }
>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    data-action-id="action_id1"
    disabled={false}
    key="action_id1"
    onClick={[Function]}
  />
</ButtonToolbar>
`;

exports[`components/action_button/ActionButton should match snapshot with hasVoted 1`] = `
<ButtonToolbar
  bsClass="btn-toolbar"
  style={
    Object {
      "marginLeft": 0,
    }
  }
>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="primary"
    data-action-id="action_id1"
    disabled={false}
    key="action_id1"
    onClick={[Function]}
  />
</ButtonToolbar>
`;
